src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java

```diff
@@ -144,6 +149,8 @@ public void testFailed(ExtensionContext context, Throwable cause) {
       }
     } else {
       firstTestPassed = false;
+      // If the first test without mutations fails, then we DO throw an exception because it should be successful
+      throw new Exception("Test failed");
       // TODO: if the first test does not pass, then we should not run it again with the mutations because it's useless
```

Comment:
Is this still needed?

Reply:
yes

Reply:
Yeah, so I guess the option here is to detect if the first test failed, set a variable and then not generate and run the mutants in this case. Let's have it as a "nice to have" follow on.